### DISCUSSION TITLE
How to replace the entire content?

### DISCUSSION THREAD

**Discussion by mattiaz9** [ianstormtaylor/slate#5157] [Q&A] [answered]
I need to to replace the editor content with a new one.

I tried setting `editor.children = newContent`, but it doesn't work.

I also tried using the `Transforms` class like this:
```ts
const newValue = JSON.parse(val) as any[]
editorValue.current.forEach((_, i) => {
  Transforms.removeNodes(editor, { at: [i] })
})
newValue.forEach((node, i) => {
  Transforms.insertNodes(editor, node, { at: [i] })
})
```

but i get the error: `Cannot find a descendant at path [0] in node: ...`

**Comment by edrdoprz** [comment]
not sure if this is the "proper" way, but it seems to have worked during my testing.

```
// clear current editor
Transforms.delete(editor, {
  at: {
    anchor: Editor.start(editor, []),
    focus: Editor.end(editor, []),
  },
})

// add children back
editor.children = newChildren
```

**Comment by mattiaz9** [comment, marked as answer]
I managed to do it in this way:

```ts
editor.replaceContent = newContent => {
    editor.children = newContent
    editor.normalizeNode([editor, []])
}
```

**Comment by VatsalDave2102** [comment]
Hello, I am also facing a similar situation to replace the editor's content but I am not able to do with this solution. I have my `newContent` stored in `localStorage` and I am using redux toolkit's reducer to replace the content.
